### PR TITLE
1：修复auth_oidc报错 2：新的自动生成镜像

### PR DESCRIPTION
**Pull request by appotry**

### CODE REVIEW

File: .github/workflows/build-release.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ name: "[builder] CI for releases"
 on:
   repository_dispatch:
     types: build
+  watch:
+    types: [started]
```
Comment (TonyRL) [inline .github/workflows/build-release.yml:7]: Unnecessary changes

File: .github/workflows/build-release.yml [hunk 2]
```diff
@@ -18,29 +20,34 @@ jobs:
...
+      - name: Set Version
+        id: set-version
         run: |
-          docker buildx build \
-          --platform=linux/amd64,linux/arm/v7,linux/arm64 \
-          --output "type=image,push=true" \
-          --file ./Dockerfile . \
-          --tag $(echo "${DOCKER_USERNAME}" | tr '[:upper:]' '[:lower:]')/ttrss:${{ steps.tag.outputs.branch }} \
-          --tag $(echo "${DOCKER_USERNAME}" | tr '[:upper:]' '[:lower:]')/ttrss:${{ steps.tag.outputs.branch }}-${{ steps.date.outputs.date }}
+          echo "version=`date +'%y-%m-%d'`" >> $GITHUB_ENV
```
Comment (TonyRL) [inline .github/workflows/build-release.yml:41]: It is recommended to utilize [docker/metadata-action](https://github.com/docker/metadata-action) from docker to sanitize inputs and correctly tag the images.

File: .github/workflows/build-release.yml [hunk 3]
```diff
@@ -18,29 +20,34 @@ jobs:
...
+      # Build and push
+      - name: Build dockerfile
+        uses: docker/build-push-action@v4
+        with:
+          file: ./Dockerfile
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
+          # platforms: linux/amd64
+          push: true
+          tags: |
+            wangqiru/ttrss:stable-${{ env.version }}
```
Comment (TonyRL) [inline .github/workflows/build-release.yml:52]: Adding an undocumented tag may confuse users. 

It is suggested to maintain consistency with the current tagging convention, which includes a `latest` tag and a dated `latest` tag format such as `latest-2023-01-23`.

**Review by TonyRL — COMMENTED** [review]
Duplicate of https://github.com/HenryQW/Awesome-TTRSS/pull/447

It's also suggested to split this PR into two separate PRs.

**Comment by DDSDerek** [comment]
At present, there will be a new error when installing Auth OIDC. You need to log in to download the file. I can't solve it.
```bash
#26 1.954 gzip: invalid magic
#26 1.955 tar: Child died with signal 13
#26 1.955 tar: Error is not recoverable: exiting now
```
```
You need to sign in or sign up before continuing.
```

File: .github/workflows/build-release.yml (hunk 1 above)
Comment (appotry) [reply .github/workflows/build-release.yml:7]: This is a very useful method to debug Docker.

File: .github/workflows/build-release.yml (hunk 2 above)
Comment (appotry) [reply .github/workflows/build-release.yml:41]: I debugged another version that uses docker/metadata-action, and the test indeed has many more features than the original one.

https://github.com/appotry/Awesome-TTRSS/blob/main/.github/workflows/build-release.yml

**Comment by appotry** [comment]
Using Git to download directly is more concise and convenient.

```
## Auth OIDC
RUN git clone --depth 1
https://gitlab.tt-rss.org/tt-rss/plugins/ttrss-auth-oidc.git auth_oidc
```

DDSDerek ***@***.***> 于2023年4月8日周六 09:21写道：

File: .github/workflows/build-release.yml [hunk 4]
```diff
@@ -3,6 +3,12 @@ name: "[builder] CI for releases"
 on:
   repository_dispatch:
     types: build
+  release:
+    types: [published]
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    # UTC 17:00 -> CST (China) 1:00, see https://datetime360.com/cn/utc-cst-china-time/
+    - cron: '0 17 * * *'
```
Comment (TonyRL) [inline .github/workflows/build-release.yml:11]: It doesn't seem like a good idea to release another nightly build.
https://github.com/HenryQW/Awesome-TTRSS/blob/e8006b2c373ba028ddd6b7df0a9b3c570ad880d5/.github/workflows/build-nightly-release.yml#L1-L5

File: .github/workflows/build-release.yml [hunk 5]
```diff
@@ -18,29 +24,40 @@ jobs:
...
-          --tag $(echo "${DOCKER_USERNAME}" | tr '[:upper:]' '[:lower:]')/ttrss:${{ steps.tag.outputs.branch }} \
-          --tag $(echo "${DOCKER_USERNAME}" | tr '[:upper:]' '[:lower:]')/ttrss:${{ steps.tag.outputs.branch }}-${{ steps.date.outputs.date }}
+        uses: docker/setup-buildx-action@v2
+
+      # https://github.com/docker/metadata-action
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: wangqiru/ttrss
```
Comment (TonyRL) [inline .github/workflows/build-release.yml:40]:
```suggestion
          images: ${{ secrets.DOCKER_USERNAME }}/ttrss
```

File: .github/workflows/build-release.yml [hunk 6]
```diff
@@ -3,6 +3,12 @@ name: "[builder] CI for releases"
 on:
   repository_dispatch:
     types: build
+  release:
+    types: [published]
```
Comment (TonyRL) [inline .github/workflows/build-release.yml:7]: Please note that this event will only be triggered if there's a new release in https://github.com/HenryQW/Awesome-TTRSS/releases

Comment (appotry) [reply .github/workflows/build-release.yml:7]: I know about release events, but the project has never used them. It's better to use nightly builds as the testing branch for development, and tags and release events for final user releases. I've encountered many instances where nightly builds failed to start properly in the past.
